sql/analyzer/validation_rules.go

```diff
@@ -512,12 +456,59 @@ func validateIntervalUsage(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Sco
 	})
 
 	if invalid {
-		return nil, transform.SameTree, ErrIntervalInvalidUse.New()
+		return nil, transform.SameTree, analyzererrors.ErrIntervalInvalidUse.New()
 	}
 
 	return n, transform.SameTree, nil
 }
 
+func validateStarExpressions(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
+	var err error
+	transform.Inspect(n, func(n sql.Node) bool {
+		if n == nil {
+			return false
+		}
+
+		if plan.IsDDLNode(n) {
+			return false
+		}
+
+		if er, ok := n.(sql.Expressioner); ok {
+			for _, e := range er.Expressions() {
+				// An expression consisting of just a * is allowed.
+				if _, s := e.(*expression.Star); s {
+					return false
+				}
+				// Otherwise, * can only be used inside acceptable aggregation functions.
+				// Detect any uses of * outside such functions.
+				sql.Inspect(e, func(e sql.Expression) bool {
```

Comment:
i didn't even realize we had an sql.Inspect

Reply:
We do! It's pretty much the same as transform.InspectExpressions except it doesn't have to start from the root of the expression tree. Useful here because we're fine with a star expression if it's the root.